core/src/main/scala/cats/Parallel.scala

```diff
@@ -97,9 +107,13 @@ object Parallel extends ParallelArityFunctions {
     */
   def parAp2[M[_]: Monad, F[_], A, B, Z](ff: M[(A, B) => Z])
                                         (ma: M[A], mb: M[B])
-                                        (implicit P: Parallel[M, F]): M[Z] = P.sequential.apply(
-    P.applicative.ap2(P.parallel.apply(ff))(P.parallel.apply(ma), P.parallel.apply(mb))
-  )
+                                        (implicit P: Parallel[M, F]): M[Z] = {
+    implicit val M = P.monad
+    P.sequential.apply(
```

Comment:
`P.sequential()`? Here and in other places, and with `P.parallel`.

Reply:
+1